db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -559,6 +559,11 @@ class NonBatchedOpsStressTest : public StressTest {
     std::vector<Slice> keys;
     key_str.reserve(num_keys);
     keys.reserve(num_keys);
+    // When Flags_use_txn is enabled, we also do a read your write check for
+    // Puts and Deletes, but not for Merges.
+    std::vector<std::string> read_your_write_values;
```

Comment:
Will be landing mine soon and after that I can help out with your rebase if needed for sure!

Reply:
Thanks for the update. I will rebase and use your new utils.

Reply:
Hey @jowlyzhang  - let's sync up on this change as depending on what you want to do with TestMultiGet, it might or might not concern ExpectedState (hence any of my PR change). As you see, prior to your PR (even post my PR), the TestMultiGet doesn't do any verification against ExpectedState like TestGet https://github.com/facebook/rocksdb/blob/main/db_stress_tool/no_batched_ops_stress.cc#L514-L547. All it cares about in TestMultiGet is the status of ops and Get() consistent with MultiGet(). I'm not sure yet what is the scope of your change and whether it needs to concern ExpectedState (hence any of my latest change) so let's talk about this. 

cc @ajkr

Reply:
Hui and I synced offline to discuss if the usage of `ExpectedValue` in this PR follow its best practice. I agreed that this usage doesn't take advantage of the full capabilities of `ExpectedValue` and just used its value base part. I think it's still an improvement over using `int32_t` value base directly, because of the nice syntax and to be consistent with other value verifications. @ajkr let us know what you think.